pyangbind/lib/serialise.py

```diff
@@ -164,54 +160,271 @@ def default(self, obj, mode="default"):
+        return float(obj)
+
+    def yangt_empty(self, obj):
+        return bool(obj)
+
+    def yangt_typed_list(self, obj):
+        return [self.default(i) for i in obj]
+
+
+class IETFYangDataSerialiser(YangDataSerialiser):
+    """
+    IETF data serialiser overrides some of the data type formats only
+    """
+
+    def yangt_long(self, obj):
+        return six.text_type(obj)
+
+    def yangt_identityref(self, obj):
+        try:
+            emod = obj._enumeration_dict[obj]["@module"]
+            if emod != obj._defining_module:
+                return "%s:%s" % (obj._enumeration_dict[obj]["@module"], obj)
+        except KeyError:
+            pass
+        return six.text_type(obj)
+
+    def yangt_decimal(self, obj):
+        return six.text_type(obj)
+
+    def yangt_empty(self, obj):
+        return [None] if obj else False
+
+
+class XmlYangDataSerialiser(IETFYangDataSerialiser):
+    """
+    XML can have an empty tag, and a TypedList must be treated specially, here we mark it with a deque
+    """
+
+    def yangt_typed_list(self, obj):
+        return deque([self.default(i) for i in obj])
+
+    def yangt_empty(self, obj):
+        return None
+
+
+class pybindJSONEncoderBase(json.JSONEncoder):
+    serialiser_klass = None
+
+    def encode(self, obj):
+        return json.JSONEncoder.encode(self, self.serialiser_klass().preprocess_element(obj))
+
+    def default(self, obj):
+        return self.serialiser_klass().default(obj)
+
+
+class pybindJSONEncoder(pybindJSONEncoderBase):
+    serialiser_klass = YangDataSerialiser
+
+
+class pybindIETFJSONEncoder(pybindJSONEncoderBase):
+    serialiser_klass = IETFYangDataSerialiser
+
+    @staticmethod
+    def yname_ns_func(parent_namespace, element, yname):
+        if not element._namespace == parent_namespace:
+            # if the namespace is different, then precede with the module
+            # name as per spec.
+            return "%s:%s" % (element._defining_module, yname)
+        else:
+            return yname
+
+    @staticmethod
+    def generate_element(obj, parent_namespace=None, flt=False, with_defaults=None):
+        ietf_tree_json_func = make_generate_ietf_tree(pybindIETFJSONEncoder.yname_ns_func)
+        return ietf_tree_json_func(obj, parent_namespace=parent_namespace, flt=flt, with_defaults=with_defaults)
+
+
+class EMF(objectify.ElementMaker):
+    """
+    Override ElementMaker to do netconf specific niceties, e.g.
+     - set the namespace
+     - convert underscore to hyphens for tags
+    """
+
+    def __init__(self, namespace=None, nsmap=None):
+        assert namespace or nsmap, "Must set either namespace or nsmap"
+        if namespace:
+            super(EMF, self).__init__(annotate=False, namespace=namespace, nsmap={None: namespace})
+        elif nsmap:
+            super(EMF, self).__init__(annotate=False, namespace=nsmap[None], nsmap=nsmap)
+
+    def __getattribute__(self, tag):
```

Comment:
Can you add some commentary on why this is required? You can use the `yang_name` attribute of any `YANGBaseClass` to get the real name in the module of any entity. This is preferable since there are other name transformations that can be done to the name in `safe_name`.